log4j2-propagation/src/test/java/nl/talsmasoftware/context/log4j2/threadcontext/Log4J2ThreadContextSnapshotTest.java

```diff
@@ -193,92 +196,55 @@ void testApplyToCurrentThread() {
         ThreadContext.push("stack1");
         ThreadContext.push("stack2");
 
-        Log4j2ThreadContextData data = Log4j2ThreadContextData.fromCurrentThreadContext();
+        Log4j2ThreadContextSnapshot snapshot = Log4j2ThreadContextSnapshot.captureFromCurrentThread();
 
         ThreadContext.clearAll();
         ThreadContext.put("map1", "old-value1");
         ThreadContext.put("map3", "old-value3");
         ThreadContext.push("stack3");
 
-        Log4j2ThreadContextData.applyToCurrentThread(data, false);
+        snapshot.applyToCurrentThread();
 
         Map<String, String> expectedMap = new HashMap<String, String>();
         expectedMap.put("map1", "value1"); // old-value1 should have been overwritten
         expectedMap.put("map2", "value2");
         expectedMap.put("map3", "old-value3");
-        assertEquals(expectedMap, ThreadContext.getContext());
+        assertThat(ThreadContext.getContext(), equalTo(expectedMap));
 
         List<String> expectedStack = Arrays.asList("stack3", "stack1", "stack2");
-        assertEquals(expectedStack, ThreadContext.getImmutableStack().asList());
+        assertThat(ThreadContext.getImmutableStack().asList(), equalTo(expectedStack));
     }
 
-    /**
-     * Verify that {@link Log4j2ThreadContextData#applyToCurrentThread(Log4j2ThreadContextData, boolean)}
-     * with {@code overwrite=true} overwrites all existing data.
-     */
     @Test
-    void testApplyToCurrentThread_overwrite() {
-        ThreadContext.put("map1", "value1");
-        ThreadContext.put("map2", "value2");
-        ThreadContext.push("stack1");
-        ThreadContext.push("stack2");
+    void testImmutableContextMap() {
```

Comment:
This test method seems to overlap with `test_umodifiableViews()` maybe it would be good to remove one of them?